storage/storage_dest.go

```diff
@@ -763,7 +809,7 @@ func (s *storageImageDestination) commitLayer(index int, info addedLayerInfo, si
 	// TODO: This can take quite some time, and should ideally be cancellable using ctx.Done().
 	layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, &storage.LayerOptions{
 		OriginalDigest:     info.digest,
-		UncompressedDigest: diffIDOrTOCDigest,
+		UncompressedDigest: layerID,
```

Comment:
Also mixing the namespaces.

Reply:
same here, we should not get so far if a partial pull is used.